akka-docs/src/main/paradox/typed/cluster-sharding.md

```diff
@@ -162,10 +162,32 @@ The `number-of-shards` configuration value must be the same for all nodes in the
 configuration check when joining. Changing the value requires stopping all nodes in the cluster.
 
 The shards are allocated to the nodes in the cluster. The decision of where to allocate a shard is done
-by a shard allocation strategy. The default implementation `akka.cluster.sharding.ShardCoordinator.LeastShardAllocationStrategy`
+by a shard allocation strategy. The default implementation @apidoc[ShardCoordinator.LeastShardAllocationStrategy]
 allocates new shards to the `ShardRegion` (node) with least number of previously allocated shards.
 This strategy can be replaced by an application specific implementation.
+
+### External shard allocation
+
+An alternative allocation strategy is the @apidoc[ExternalShardAllocationStrategy] which allows
+explicit control over where shards are allocated via the @apidoc[ExternalShardAllocation] extension.
+This can be used, for example, to match up Kafka Partition consumption with shard locations.
+
+To use it set it as the allocation strategy on your `Entity`:
+
+@@snip [ExternalShardAllocationCompileOnlySpec](/akka-cluster-sharding-typed/src/test/scala/docs/akka/cluster/sharding/typed/ExternalShardAllocationCompileOnlySpec.scala) { #entity }
+
+For any shardId that has not been allocated it will be allocated to the requesting node. To make explicit allocations:
    
+@@snip [ExternalShardAllocationCompileOnlySpec](/akka-cluster-sharding-typed/src/test/scala/docs/akka/cluster/sharding/typed/ExternalShardAllocationCompileOnlySpec.scala) { #client }
```

Comment:
shouldn't those two snippets be scala and java?

Reply:
Doh, had done the code examples just forgot to add snippet